External/Plugins/ASCompletion/Completion/ASGenerator.cs

```diff
@@ -2815,12 +2816,114 @@ private static void GenerateFunctionJob(GeneratorJobType job, ScintillaControl s
+                {
+                    parCount = 0;
+                    braCount = 0;
+                    genCount = 0;
+                    var s = caller.Parameters[parameterIndex].Type;
+                    var startPosition = 0;
+                    var arrowLength = "->".Length;
+                    var endPosition = s.LastIndexOf("->") + arrowLength;
+                    for (var i = 0; i < endPosition; i++)
+                    {
+                        string type = null;
+                        var c = s[i];
+                        if (c == '(') parCount++;
+                        else if (c == ')')
+                        {
+                            parCount--;
+                            if (parCount == 0 && braCount == 0 && genCount == 0)
+                            {
+                                i++;
+                                type = s.Substring(startPosition, i - startPosition);
+                            }
+                        }
+                        else if (c == '{') braCount++;
+                        else if (c == '}')
+                        {
+                            braCount--;
+                            if (parCount == 0 && braCount == 0 && genCount == 0)
+                            {
+                                i++;
+                                type = s.Substring(startPosition, i - startPosition);
+                            }
+                        }
+                        else if (c == '<') genCount++;
+                        else if (c == '>' && s[i - 1] != '-')
+                        {
+                            genCount--;
+                            if (parCount == 0 && braCount == 0 && genCount == 0)
+                            {
+                                i++;
+                                type = s.Substring(startPosition, i - startPosition);
+                            }
+                        }
+                        else if (parCount == 0 && braCount == 0 && genCount == 0 && c == '-' && s[i + 1] == '>')
+                        {
+                            type = s.Substring(startPosition, i - startPosition);
+                            i += arrowLength;
+                        }
+                        if (type != null)
+                        {
+                            var parameter = $"parameter{functionParameters.Count}";
+                            if (type.StartsWith('?'))
+                            {
+                                parameter = $"?{parameter}";
+                                type = type.TrimStart('?');
+                            }
+                            functionParameters.Add(new FunctionParameter(parameter, type, type, callerExpr));
+                            startPosition = i;
+                        }
+                    }
+                    newMemberType = s.Substring(endPosition);
+                }
+            }
```

Comment:
Some notes:
- It would make sense to have this extracted outside in a reusable way so others could also parse function types too. 
- I have the feeling `while (pos > 0)` can be optimized. Although it should never reach the exit condition I think.
- This wouldn't cover cases with generics like `function test<K>(arg:K->String):Void`, where you would like to create a new function also with the `<K>` type parameter, but step by step.

Anyway, good to merge I would say.